coriolis/tests/scheduler/rpc/test_client.py

```diff
@@ -0,0 +1,241 @@
+
+        result = self.client.get_any_worker_service(
+            mock.sentinel.ctxt, random_choice=True)
+
+        self.assertIsInstance(result, dict)
+
+    @mock.patch.object(client.SchedulerClient, '_call')
+    def test_get_worker_service_for_specs(self, mock_call):
+        ctxt = mock.sentinel.ctxt
+        provider_requirements = mock.sentinel.provider_requirements
+        region_sets = mock.sentinel.region_sets
+        enabled = mock.sentinel.enabled
+        raise_on_no_matches = mock.sentinel.raise_on_no_matches
+
+        self.client.get_worker_service_for_specs(
+            ctxt, provider_requirements=provider_requirements,
+            region_sets=region_sets, enabled=enabled,
+            raise_on_no_matches=raise_on_no_matches)
+
+        mock_call.assert_called_once_with(
+            ctxt, 'get_workers_for_specs', region_sets=region_sets,
+            enabled=enabled, provider_requirements=provider_requirements)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_worker_service_for_specs_no_services_no_raise(
+            self, mock_get_workers):
+        mock_get_workers.return_value = []
+        result = self.client.get_worker_service_for_specs(
+            mock.sentinel.ctxt, raise_on_no_matches=False)
+        self.assertIsNone(result)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_worker_service_for_specs_no_services(self, mock_get_workers):
+        mock_get_workers.return_value = []
+        self.assertRaises(
+            exception.NoSuitableWorkerServiceError,
+            self.client.get_worker_service_for_specs, mock.sentinel.ctxt)
+
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    @mock.patch('random.choice')
+    def test_get_worker_service_for_specs_random_choice(
+            self, mock_random_choice, mock_get_workers):
+        service_mock1 = {'id': 'test_id1'}
+        service_mock2 = {'id': 'test_id2'}
+        mock_get_workers.return_value = [service_mock1, service_mock2]
+        mock_random_choice.return_value = service_mock1
+
+        result = self.client.get_worker_service_for_specs(
+            mock.sentinel.ctxt, random_choice=True)
+
+        mock_random_choice.assert_called_once_with([
+            service_mock1, service_mock2])
+        mock_get_workers.assert_called_once_with(
+            mock.sentinel.ctxt, provider_requirements=None, region_sets=None,
+            enabled=True)
+        self.assertEqual(result, service_mock1)
+
+    @mock.patch('random.choice')
+    @mock.patch.object(client.SchedulerClient, 'get_workers_for_specs')
+    def test_get_worker_service_for_specs_no_random_choice(
+            self, get_workers_for_specs, mock_random_choice):
+        mock_service = mock.MagicMock()
+        get_workers_for_specs.return_value = [mock_service]
+        mock_random_choice.return_value = mock_service
```

Comment:
We don't care about `random`'s return value here. Please add a `mock_random_choice.assert_not_called()` instead